### PR TITLE
feat(v2): improve @docusaurus/plugin-content-blog

### PR DESCRIPTION
**Pull request by endiliey**
## Motivation

Improve @docusaurus/plugin-content-blog plugin to be more like v1. There are some room for improvements (sidebar, CSS, etc). Maybe future PR will improve that.

Others:
- Components should be part of plugin package. Not in docusaurus own package
- Fix some minor bugs

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

- `/blog` now display truncated content only with readmore button

Before
image

After
blogpage

- BlogPost
blogpost

- https://deploy-preview-1311--docusaurus-2.netlify.com/blog

### CODE REVIEW

File: packages/docusaurus-plugin-content-blog/components/BlogPost/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import React from 'react';
+import Head from '@docusaurus/Head';
+import Layout from '@theme/Layout'; // eslint-disable-line
+
+import DocusaurusContext from '@docusaurus/context';
+import Post from '../Post';
+import styles from './styles.module.css';
+
+class BlogPost extends React.Component {
```
Comment (yangshun) [inline packages/docusaurus-plugin-content-blog/components/BlogPost/index.js]: Use a functional component?

File: packages/docusaurus-plugin-content-blog/components/BlogPost/styles.module.css [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ */
+
+.blogPostContainer {
+  border-bottom: 1px solid #e0e0e0;
+    border-radius: 3px;
+    padding-bottom: 20px;
+    margin: 50px auto;
+    text-align: left;
+    max-width: 1100px;
+}
```
Comment (yangshun) [inline packages/docusaurus-plugin-content-blog/components/BlogPost/styles.module.css]: Add new line  at end of files. (probably configure your editor to do it).
This file is also mis-indented.

File: packages/docusaurus-plugin-content-blog/components/Post/styles.module.css [hunk 3]
```diff
@@ -45,3 +45,38 @@
...
+
+.postContainer {
+  border-bottom: 1px solid #e0e0e0;
+    border-radius: 3px;
+    padding-bottom: 20px;
+    margin: 50px auto;
+    text-align: left;
+    width: 600px;
+    max-width: 1100px;
+}
```
Comment (yangshun) [inline packages/docusaurus-plugin-content-blog/components/Post/styles.module.css]: Ditto

**Comment by yangshun** [comment]
Btw I really like this Blog layout! Heads up that this might change in future though, the design is not finalized yet.

File: packages/docusaurus-plugin-content-blog/components/BlogPost/styles.module.css (hunk 2 above)
Comment (endiliey) [reply packages/docusaurus-plugin-content-blog/components/BlogPost/styles.module.css]: Oh my bad. I usually rely on prettier. Just realize we had no prettier setup on `.css`

**Comment by endiliey** [comment]
I think I'll leave the UI part for now and focus on the logic part for now. We had a UI framework for v2, right ?

**Comment by endiliey** [comment]
@yangshun 

> I was thinking of making the various blog components part of the UI framework instead and content plugins are only in-charge of parsing and combining data. The plugin wraps components and pass children components the necessary data but is not in-charge of rendering the layout.

Sure. I think that will be nice and good.

**Review by Joshualov — COMMENTED** [review]
‡ª"´0+&*`E}ªá‹¾~ïÌ?ÎÿÊºbÊo"œö ä^<	‚ÌÌÝ¹ãy��?¯|‰úÞ+r£|0BØZïnuoJA¯ež	Ÿ¦�ëŸ:K6­ÙQQª¥Úimi4ÖÎ+´ÅU~�Ü²á±áä%ª€ÊãðglüL�ÜÐrzŒniBQ›ØqÿU&7ÅQpôƒÂ?¢a"ÏQrÖ	“¾äÅÄ‰yÈ°BÛÙ¿´b¸§,p§Ê@_¬AD7<Zx�<&¤�{•›w–.É´qÊFói+CÇ$b%_¦™ð@½aö/Ã9<ËYwy¡R=ùFµ%éP‚m�—ä™s%§èÁ×&àöZp¸ W¶U^ �û0{”sô‰í‹.ášò:•tdžfÕÄ¨´î1âv»½ 
 ‡ª"´0 &*`E}ªá‹¾~ïÌ?ÎÿÊºbÊo"œö ä^<‚ÌÌÝ¹ãy��?¯|

**Pull request merged**